mlir/lib/Target/LLVMIR/ModuleTranslation.cpp

```diff
@@ -446,6 +447,90 @@ convertDenseElementsAttr(Location loc, DenseElementsAttr denseElementsAttr,
   return buildSequentialConstant(constantsRef, outerShape, llvmType, loc);
 }
 
+/// Convert a dense resource elements attribute to an LLVM IR constant using its
+/// raw data storage if possible. This supports elements attributes of tensor or
+/// vector type and avoids constructing separate objects for individual values
+/// of the innermost dimension. Constants for other dimensions are still
+/// constructed recursively. Returns null if constructing from raw data is not
+/// supported for this type, e.g., element type is not a power-of-two-sized
+/// primitive. Reports other errors at `loc`.
```

Comment:
Since there is a fallback, it's not clear to me about the error reporting logic?

Reply:
Errors at loc are emitted (using emitError) by buildSequentialConstant along with the fallback.

Reply:
My point is: either it's an error, and we should propagate and abort the process, or there is a fallback and we should not emitError...

Reply:
Right. I'll can remove the emitError from buildSequentialConstant.

Reply:
Instead of removing this, the function now emits an error, returns nullptr, which is propagated above. So there is no fallback now. There is no other current conversion path for DenseResourceElementsAttr, so this is correct behavior.